### PR TITLE
response comparer: add feature that ignore sort order when same sort key

### PR DESCRIPTION
**Pull request by komainu8**

### CODE REVIEW

**Comment by komainu8** [comment]
@kenhys Could you review this PR, please?

File: lib/groonga-query-log/response-comparer.rb [hunk 1]
```diff
@@ -59,16 +59,61 @@ def same_response?
       @response1.body == @response2.body
     end
 
+    def same_sort_key?
+      return false unless @command.has_key?(:sort_keys)
+      if !@command.sort_keys.empty?
+        @response1.records.each do |record1|
```
Comment (kenhys) [inline lib/groonga-query-log/response-comparer.rb]:
```
return false if @command.sort_keys.empty?
@response1.records.each do |...|
...
end
return true
```
It seems that reducing nested block make the code simple.

File: lib/groonga-query-log/response-comparer.rb [hunk 2]
```diff
@@ -59,16 +59,61 @@ def same_response?
...
+        if compare_result
+          compare_result = same_drilldowns?
+        end
+        if compare_result == false
+          if same_sort_key? or same_shard_key?
+            return same_size_response?
+          end
         end
-        same_drilldowns?
+        compare_result
```
Comment (kenhys) [inline lib/groonga-query-log/response-comparer.rb]:
```
if compare_result
...
else
...
end
```
It seems that `if ... else ... ` is simple enough. (instead of if compare_result ... if compare_result == false)

File: lib/groonga-query-log/response-comparer.rb [hunk 3]
```diff
@@ -59,16 +59,61 @@ def same_response?
...
+        return true
+      else
+        return false
+      end
+    end
+
+    def same_shard_key?
+      return false unless @command.has_key?(:shard_key)
+      if !@command.shard_key.empty?
+        @response1.records.each do |record1|
```
Comment (kenhys) [inline lib/groonga-query-log/response-comparer.rb]: `return false if @command.shard_key.empty? ` 
It seems better to return in early stage.(it also reduces nested block)

File: test/test-response-comparer.rb [hunk 4]
```diff
@@ -183,6 +183,30 @@ def test_minus
             assert_true(score_sort?("_id,-_score,_key"))
           end
         end
+
+        class SameSortKeyTest < self
+          def test_same_sort_key
+            @command[:sort_keys] = "_score"
```
Comment (kenhys) [inline test/test-response-comparer.rb]: It seems that same shard key test case is missing.

File: lib/groonga-query-log/response-comparer.rb [hunk 5]
```diff
@@ -59,16 +59,55 @@ def same_response?
       @response1.body == @response2.body
     end
 
+    def same_sort_key?
+      return false unless @command.has_key?(:sort_keys)
+      return false if @command.sort_keys.empty?
```
Comment (kou) [inline lib/groonga-query-log/response-comparer.rb]:
```suggestion
      sort_keys = @command.sort_keys
      return false if sort_keys.empty?
```

File: lib/groonga-query-log/response-comparer.rb [hunk 6]
```diff
@@ -59,16 +59,55 @@ def same_response?
       @response1.body == @response2.body
     end
 
+    def same_sort_key?
+      return false unless @command.has_key?(:sort_keys)
```
Comment (kou) [inline lib/groonga-query-log/response-comparer.rb]: This is not good. It doesn't work with `sortby`.

`@command.sort_keys` returns `sort_keys` or `sortby` and returns `[]` for no `sort_keys`/`sortby` case.

File: lib/groonga-query-log/response-comparer.rb [hunk 7]
```diff
@@ -59,16 +59,54 @@ def same_response?
...
+      return false if @command.sort_keys.empty?
+      @response1.records.each do |record1|
+        @response2.records.each do |record2|
+          if record2[@command.sort_keys] == record1[@command.sort_keys]
```
Comment (kou) [inline lib/groonga-query-log/response-comparer.rb]: `return false unless record2[@command.sort_keys] == record1[@command.sort_keys]`

File: test/test-response-comparer.rb [hunk 8]
```diff
@@ -588,3 +612,5 @@ def test_with_location
     end
   end
 end
+
+
```
Comment (kou) [inline test/test-response-comparer.rb]: Could you revert this change?

**Comment by komainu8** [comment]
@kou Thank you for your comments.
I reverted my modify.
I remake this feature. Could you review again?
